src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3147,49 +3274,99 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
+            Vec<char*> member_fn_names;
+            Vec<ASR::call_arg_t> member_init;
+            member_names.reserve(al, x.n_body);
+            member_fn_names.reserve(al, 1);
+            member_init.reserve(al, x.n_body);
+            SetChar struct_dependencies;
+            struct_dependencies.reserve(al, 1);
+            ASR::abiType class_abi = ASR::abiType::Source;
+            if( is_bindc_class(x.m_decorator_list, x.n_decorator_list) ) {
+                class_abi = ASR::abiType::BindC;
+            }
+            visit_ClassMembers(x, member_names, member_fn_names, struct_dependencies, member_init, false, class_abi);//TODO add member functions to the list
+            LCOMPILERS_ASSERT(member_init.size() == member_names.size());
+            ASR::symbol_t* class_type = ASR::down_cast<ASR::symbol_t>(ASR::make_Struct_t(al,
+                                            x.base.base.loc, current_scope, x.m_name,
+                                            struct_dependencies.p, struct_dependencies.size(),
+                                            member_names.p, member_names.size(),
+                                            member_fn_names.p, member_fn_names.size(),
+                                            class_abi, ASR::accessType::Public,
+                                            is_packed, false, member_init.p, member_init.size(), aligned_expr,
+                                            nullptr));
+            current_scope = parent_scope;
+            if (current_scope->resolve_symbol(x_m_name)) {
+                ASR::symbol_t* sym = current_scope->resolve_symbol(x_m_name);
+                ASR::Struct_t *st = ASR::down_cast<ASR::Struct_t>(sym);
+                st->m_initializers = member_init.p;
+                st->n_initializers = member_init.size();
+            } else {
+                current_scope->add_symbol(x_m_name, class_type);
+            }
         } else {
-            current_scope->add_symbol(x_m_name, class_type);
+            if( x.n_bases > 0 ) {
+                throw SemanticError("Inheritance in classes isn't supported yet.",
+                                    x.base.base.loc);
+            }
+            SymbolTable *parent_scope = current_scope;
+            ASR::symbol_t* sym = current_scope->resolve_symbol(x_m_name);
+            if( sym && !(ASR::is_a<ASR::Variable_t>(*sym) &&
+                ASR::is_a<ASR::TypeParameter_t>(*ASR::down_cast<ASR::Variable_t>(sym)->m_type))) {
+                LCOMPILERS_ASSERT(ASR::is_a<ASR::Struct_t>(*sym));
+                ASR::Struct_t *st = ASR::down_cast<ASR::Struct_t>(sym);
+                current_scope = st->m_symtab;
+                for( size_t i = 0; i < x.n_body; i++ ) {
+                    if ( AST::is_a<AST::FunctionDef_t>(*x.m_body[i]) ) {
+                        AST::FunctionDef_t* f = AST::down_cast<AST::FunctionDef_t>(x.m_body[i]);
+                        if ( std::string(f->m_name) != std::string("__init__") ){
```

Comment:
Minor: style suggestion, also update everywhere.
```suggestion
                        if ( std::string(f->m_name) != std::string("__init__") ) {
```